pkg/autodiscovery/listeners/types.go

```diff
@@ -14,14 +14,20 @@ import (
 // ID is the representation of the unique ID of a Service
 type ID string
 
+// ContainerPort represents a network port in a Service.
+type ContainerPort struct {
+	Port int
```

Comment:
I can make this `PortNumber` if preferred, just seems overly verbose to add `Number` to an int field. Naming is hard.